src/hotspot/share/runtime/flags/jvmFlag.cpp

```diff
@@ -701,26 +701,28 @@ void JVMFlag::printFlags(outputStream* out, bool withComments, bool printRanges,
     out->print_cr("[Global flags ranges]");
   }
 
-  // Sort
-  JVMFlag** array = NEW_C_HEAP_ARRAY_RETURN_NULL(JVMFlag*, length, mtArguments);
-  if (array != nullptr) {
-    for (size_t i = 0; i < length; i++) {
-      array[i] = &flagTable[i];
+  // Mark flags clear
+  for (size_t i = 0; i < length; i++) {
+    if (flagTable[i].is_unlocked() && !(skipDefaults && flagTable[i].is_default())) {
+      flagTable[i].clear_iterated();
     }
-    qsort(array, length, sizeof(JVMFlag*), compare_flags);
-
+  }
+  // Print the flag with highest sort value, then mark it
+  for (size_t j = 0; j < length; j++) {
+    JVMFlag* max = nullptr;
```

Comment:
I would consider this to be the "min" not "max"

Reply:
I will use "best" instead.